mailpoet/lib/Subscribers/SubscriberSegmentRepository.php

```diff
@@ -81,8 +81,34 @@ public function unsubscribeFromSegments(SubscriberEntity $subscriber, array $seg
   }
 
   public function resetSubscriptions(SubscriberEntity $subscriber, array $segments): void {
-    $this->unsubscribeFromSegments($subscriber);
-    $this->subscribeToSegments($subscriber, $segments);
+    $existingSegments = array_values(array_filter(array_map(
+      function(SubscriberSegmentEntity $subscriberSegmentEntity): ?SegmentEntity { return $subscriberSegmentEntity->getSegment(); 
+      },
+      $this->findBy(['subscriber' => $subscriber, 'status' => SubscriberEntity::STATUS_SUBSCRIBED])
+    )));
+    $existingSegmentIds = array_map(function(SegmentEntity $segment): int { return $segment->getId() ?? 0;
```

Comment:
The code formatting here is weird. Can you check it, please? There is an empty line below, so I think the return statement should be at the line. You can find more similar cases in this method.

Reply:
54872c01b